src/diffusers/models/unet_motion_blocks.py

```diff
@@ -0,0 +1,1234 @@
+    add_upsample,
+    resnet_eps,
+    resnet_act_fn,
+    num_attention_heads,
+    resolution_idx=None,
+    resnet_groups=None,
+    cross_attention_dim=None,
+    dual_cross_attention=False,
+    use_linear_projection=True,
+    only_cross_attention=False,
+    upcast_attention=False,
+    resnet_time_scale_shift="default",
+    motion_norm_num_groups=32,
+    motion_cross_attention_dim=None,
+    motion_num_attention_heads=8,
+    motion_attention_bias=False,
+    motion_activation_fn="geglu",
+    motion_max_seq_length=32,
+):
+    if up_block_type == "UpBlockMotion":
+        return UpBlockMotion(
+            num_layers=num_layers,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            prev_output_channel=prev_output_channel,
+            temb_channels=temb_channels,
+            add_upsample=add_upsample,
+            resnet_eps=resnet_eps,
+            resnet_act_fn=resnet_act_fn,
+            resnet_groups=resnet_groups,
+            resnet_time_scale_shift=resnet_time_scale_shift,
+            resolution_idx=resolution_idx,
+            motion_norm_num_groups=motion_norm_num_groups,
+            motion_cross_attention_dim=motion_cross_attention_dim,
+            motion_num_attention_heads=motion_num_attention_heads,
+            motion_attention_bias=motion_attention_bias,
+            motion_activation_fn=motion_activation_fn,
+            motion_max_seq_length=motion_max_seq_length,
+        )
+    elif up_block_type == "CrossAttnUpBlockMotion":
+        if cross_attention_dim is None:
+            raise ValueError("cross_attention_dim must be specified for CrossAttnUpBlockMotion")
+        return CrossAttnUpBlockMotion(
+            num_layers=num_layers,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            prev_output_channel=prev_output_channel,
+            temb_channels=temb_channels,
+            add_upsample=add_upsample,
+            resnet_eps=resnet_eps,
+            resnet_act_fn=resnet_act_fn,
+            resnet_groups=resnet_groups,
+            cross_attention_dim=cross_attention_dim,
+            num_attention_heads=num_attention_heads,
+            dual_cross_attention=dual_cross_attention,
+            use_linear_projection=use_linear_projection,
+            only_cross_attention=only_cross_attention,
+            upcast_attention=upcast_attention,
+            resnet_time_scale_shift=resnet_time_scale_shift,
+            resolution_idx=resolution_idx,
+            motion_norm_num_groups=motion_norm_num_groups,
+            motion_cross_attention_dim=motion_cross_attention_dim,
+            motion_num_attention_heads=motion_num_attention_heads,
+            motion_attention_bias=motion_attention_bias,
+            motion_activation_fn=motion_activation_fn,
+            motion_max_seq_length=motion_max_seq_length,
+        )
+    raise ValueError(f"{up_block_type} does not exist.")
+
+
+class MotionAttnProcessor(nn.Module):
```

Comment:
These attn processors are very specific to motion so let's add them to `unet_modition_model.py`

Reply:
Hmm so there might be a circular dependency issue here. 

The UNet Motion blocks have Motion Modules that use these attention processors. So you would end up in a situation where importing the blocks into `unet_motion_model.py` would require importing the model into the blocks. 

Could we create a separate `motion.py` file for the MotionModules, MotionAdapter, Motion Attn Processors and move the UNet Motion blocks into the 2D Unet blocks file?

Reply:
Ah I see, I'd then maybe move the attn processors to `attention.py` and the blocks into unet_3d_blocks